File: plugins/sonar-jacoco-plugin/src/main/java/org/sonar/plugins/jacoco/itcoverage/viewer/client/Metrics.java
/*
 * Sonar, open source software quality management tool.
 * Copyright (C) 2008-2011 SonarSource
 * mailto:contact AT sonarsource DOT com
 *
 * Sonar is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * Sonar is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with Sonar; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
 */
package org.sonar.plugins.jacoco.itcoverage.viewer.client;

/**
 * Should be in {@link org.sonar.gwt.Metrics}
 */
public interface Metrics {
  String IT_COVERAGE = "it_coverage";
  String IT_LINES_TO_COVER = "it_lines_to_cover";
  String IT_UNCOVERED_LINES = "it_uncovered_lines";
  String IT_LINE_COVERAGE = "it_line_coverage";
  String IT_COVERAGE_LINE_HITS_DATA = "it_coverage_line_hits_data";
  String IT_CONDITIONS_TO_COVER = "it_conditions_to_cover";
  String IT_UNCOVERED_CONDITIONS = "it_uncovered_conditions";
  String IT_BRANCH_COVERAGE = "it_branch_coverage";
  String IT_CONDITIONS_BY_LINE = "it_conditions_by_line";
  String IT_COVERED_CONDITIONS_BY_LINE = "it_covered_conditions_by_line";

  /**
   * @deprecated use IT_CONDITIONS_BY_LINE and IT_COVERED_CONDITIONS_BY_LINE
   */
  @Deprecated
  String IT_BRANCH_COVERAGE_HITS_DATA = "it_branch_coverage_hits_data";
}

File: server/sonar-web/src/main/js/apps/overview/helpers/metrics.js
/*
 * SonarQube :: Web
 * Copyright (C) 2009-2016 SonarSource SA
 * mailto:contact AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
function hasRightDomain (metric, domains) {
  return domains.indexOf(metric.domain) !== -1;
}

function isNotHidden (metric) {
  return !metric.hidden;
}

function hasSimpleType (metric) {
  return metric.type !== 'DATA' && metric.type !== 'DISTRIB';
}

function isNotDifferential (metric) {
  return metric.key.indexOf('new_') !== 0;
}

export function filterMetrics (metrics) {
  return metrics.filter(metric => {
    return isNotHidden(metric) && hasSimpleType(metric) && isNotDifferential(metric);
  });
}

export function filterMetricsForDomains (metrics, domains) {
  return filterMetrics(metrics).filter(metric => hasRightDomain(metric, domains));
}


export function getShortType (type) {
  if (type === 'INT') {
    return 'SHORT_INT';
  } else if (type === 'WORK_DUR') {
    return 'SHORT_WORK_DUR';
  }
  return type;
}


export function getMetricName (metricKey) {
  return window.t('overview.metric', metricKey);
}
